File: javascript/src/detectors/loose_file_permissions/loose_file_permissions.js
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

// {fact rule=loose-file-permissions@v1.0 defects=1}
var fs = require('fs')
function looseFilePermissionsNoncompliant() {
    // Noncompliant: read permissions assigned to others.
    fs.promises.chmod("/path", 0o774).then(r => {})
}
// {/fact}


// {fact rule=loose-file-permissions@v1.0 defects=0}
var fs = require('fs');
function looseFilePermissionsCompliant() {
    // Compliant: read, write and execute permissions assigned to owner and no permission assigned to others.
    fs.promises.chmod("/path", 0o770).then(r => {})
}
// {/fact}

File: ruby/src/detectors/loose-file-permissions/loose-file-permissions.rb
=begin
 Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 SPDX-License-Identifier: Apache-2.0
=end

# {fact rule=loose-file-permissions@v1.0 defects=1}
require "fileutils"

def open_file_permission_noncompliant(filename)

  # Noncompliant: sets file world writable.
  FileUtils.chmod 0222, filename
end
# {/fact}

# {fact rule=loose-file-permissions@v1.0 defects=0}
def open_file_permission_compliant(filename)

  # Compliant: restricts group/world access.
  FileUtils.chmod 0700, filename
end
# {/fact}